### PR TITLE
Add robostack

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
~~Some functionality is still missing:~~
- ~~the "REQUIRE_OPENGL" and "REQUIRE_GL" hacks don't work~~
- ~~I'm not sure what happens if a version is specified in the robostack.yaml (formerly conda-forge.yaml).~~

### CODE REVIEW

File: src/rosdep2/platforms/robostack.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import subprocess
+
+from rospkg.os_detect import OS_ROBOSTACK
+
+from ..installers import PackageManagerInstaller
+
+ROBOSTACK_INSTALLER = 'robostack'
+
+
+def is_mamba_installed():
```
Comment (wolfv) [inline src/rosdep2/platforms/robostack.py:38]: we could potentially also use this one: https://github.com/conda-incubator/ensureconda

Maybe not necessary though.

File: src/rosdep2/platforms/robostack.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    context.set_installer(ROBOSTACK_INSTALLER, RoboStackInstaller())
+
+
+def register_platforms(context):
+    context.add_os_installer_key(OS_ROBOSTACK, ROBOSTACK_INSTALLER)
+    context.set_default_os_installer_key(OS_ROBOSTACK, lambda self: ROBOSTACK_INSTALLER)
+
+
+def conda_detect(packages):
+    conda_ret = subprocess.check_output(['conda', 'list', '-e', '-q']).splitlines()
```
Comment (wolfv) [inline src/rosdep2/platforms/robostack.py:56]: how about using `--json`?

**Comment by wolfv** [comment]
nice! I added some minor comments

File: src/rosdep2/platforms/robostack.py (hunk 2 above)
Comment (Tobias-Fischer) [reply src/rosdep2/platforms/robostack.py:56]: That's a good idea! Would you be okay to write the parser? You're probably 10 times faster than me with json handling ;).

File: src/rosdep2/platforms/robostack.py (hunk 1 above)
Comment (Tobias-Fischer) [reply src/rosdep2/platforms/robostack.py:38]: It would add another dependency for quite a simple task - I don't think it's necessary?

Comment (wolfv) [reply src/rosdep2/platforms/robostack.py:38]: yeah, agreed.

File: src/rosdep2/platforms/robostack.py (hunk 2 above)
Comment (wolfv) [reply src/rosdep2/platforms/robostack.py:56]: yep, sure, I can do it later today!

Comment (wolfv) [reply src/rosdep2/platforms/robostack.py:56]: actually I just had a look and I think it's fine. wasn't so aware of the `-e` flag.

**Pull request merged**